### PR TITLE
replace connection lost dialogue

### PR DESCRIPTION
**Pull request by VersBersh**
remove the dialogue that appears when a connection to the server is lost, and replace it with styling to the kernelStatusComponent in the status bar.

## References

fixes: #8374

## Code changes

- Replace the default handler of the connection lost signal with a stump method
- Add some CSS styling to the kernelStatusComponent so that critical kernel statuses (connecting, disconnected, unknown, dead) show up in bold red

## User-facing changes
before:
image

after:
image

### CODE REVIEW

File: packages/statusbar/src/defaults/kernelStatus.tsx [hunk 1]
```diff
@@ -60,6 +61,11 @@ namespace KernelStatusComponent {
      * The status of the kernel.
      */
     status?: string;
+
+    /**
+     * Style for the status line
```
Comment (blink1073) [inline packages/statusbar/src/defaults/kernelStatus.tsx:66]: Can you please make a comments sentence case with a period at the end?

**Comment by blink1073** [comment]
cc @tgeorgeux  for UX input.

**Comment by jasongrout** [comment]
Thanks for working on this. Some people hide the status bar. Perhaps the toolbar kernel status indicator should also show something? Something indicates it is disconnected and needs to connect? @tgeorgeux - thoughts?

**Comment by tgeorgeux** [comment]
What if we also flashed the warning color and bold font on the Python 3 in toolbar on the notebook, alternately we could flash the kernel status indicator if if the Python 3 is too disruptive.

**Comment by jasongrout** [comment]
What if the kernel status indicator became a red progress circle thingy when it was in the "Connecting" state (or maybe just animated once or twice to catch attention then stopped before it became annoying), and a red X or something indicating a broken link when it was disconnected?

**Comment by VersBersh** [comment]
Perhaps the kernel status indicator could be green for connected, red for disconnected, orange/yellow for busy, and grey/clear for unknown?

**Comment by jasongrout** [comment]
I think we need a shape indicator as well for those that can't distinguish between colors as well.

**Comment by VersBersh** [comment]
something like these?
image

**Comment by jasongrout** [comment]
Nice, I like the idea of putting something in the circle.

In general, when the kernel is connected, we'll know whether it is busy or idle (the exception here is when we connect to an already-running kernel, we won't know it's busy/idle until it finishes the current computation). When it is not connected, we'll have connecting or disconnected states. So I'd suggest that useful states to indicate to the user are:

Connection state if not "connected"
1. Disconnected
2. Connecting

If connection state is "connected":
3. Busy
4. Idle
5. (possibly?) starting
6. (possibly?) unknown

**Comment by VersBersh** [comment]
In that case, it might make make sense just to use `Kernel.ConnectionStatus`, which only takes the values connected, connecting, and disconnected.

**Comment by VersBersh** [comment]
For instance, the last commit make the connection status a green tick when connected, a red x while connecting/disconnected, and the empty circle when no kernel is selected.

image

although I'm not really happy with the quality of the svg or the fact that it's a different shade of red to the kernel status in the status bar.

**Comment by blink1073** [comment]
Elegant work @VersBersh, I like it!

**Comment by blink1073** [comment]
(agreed the visual could use a bit of tweaking)

**Comment by blink1073** [comment]
Can you please add an updated screenshot @VersBersh?

**Comment by VersBersh** [comment]
yep sorry, these are the three states

image

image

image

**Comment by blink1073** [comment]
I think we do need another state for busy.  @tgeorgeux any input on what that should look like?

**Comment by tgeorgeux** [comment]
I think the busy Icon makes sense to have a "-" and or a "..." inside a solid circle. 

There are Material Icons for all four of these options available for each option.
Connected: https://materialdesignicons.com/icon/checkbox-marked-circle
Disconnected: https://materialdesignicons.com/icon/close-circle
Busy: https://materialdesignicons.com/icon/minus-circle 
Starting Or Connecting: https://materialdesignicons.com/icon/dots-horizontal-circle
Unknown: https://materialdesignicons.com/icon/checkbox-blank-circle-outline

**Comment by jasongrout** [comment]
Remember that we are representing *two* statuses here. I think the circular indicator should represent the connection status when we are disconnected or connecting, but when we are connected the indicators should represent the kernel status, not the connection status. With one exception - that if the kernel status is dead, I think the indicator should represent the dead kernel even if we are disconnected. The kernel status is one of these:
https://github.com/jupyterlab/jupyterlab/blob/6d5240dda71fb3d12814e8b88b09e94e23d42875/packages/services/src/kernel/messages.ts#L514-L540

**Comment by VersBersh** [comment]
This is just my use case, but I only look at the kernel status indicator on two occasions:
    -  I want to know that I'm (re)connected. i.e. I've just opened jupyterlab or reconnected to my vpn.
    -  I’ve run a cell, the input prompt shows [*], and I want to confirm that the kernel is actually executing my code

IMO, the informative statuses would be

Icon | Meaning for user | KernelDisplayStatus
-------------------------------------------------------------
Connected | Everything is good. The kernel will execute my code immediately (or after executing my own code) | connected, busy, idle
Busy | The kernel is doing something other than executing my code, but it will execute my code soon | initializing, starting, restarting , autorestarting
No kernel | Select the kernel so the code can execute | <empty string> (I think?)
Disconnected | Something’s wrong. Check your connection | disconnected, connecting, dead
Unknown | Don’t rely on this indicator | unknown

**Comment by jasongrout** [comment]
One thought is that the ring indicates the connection status and the center of the circle indicates the kernel status.

**Comment by VersBersh** [comment]
I like it, but perhaps the icon is too small for that? Another option (with potentially controversial animation):

jlab_icon_demo

**Comment by goanpeca** [comment]
Is this PR still valid @VersBersh ?

Thanks! :) 

---

Pinging @isabela-pf

**Comment by jweill-aws** [comment]
There hasn't been any activity on this pull request in over a year. As part of an effort to clean up this repository's pull requests, I'm closing this request. If you would still like to merge this code in, please reopen the request and resolve any outstanding merge conflicts. All of the code and comments will remain in place. Thank you for being part of the JupyterLab community!

**Pull request closed without merging**